src/fsharp/PostInferenceChecks.fs

```diff
@@ -681,23 +682,75 @@ let CheckNoReraise cenv freesOpt (body: Expr) =
-                       not (typeEquivAux EraseNone cenv.g typ1 typ2) 
-                    then Some (typ1, typ2)
-                    else None))
-    match firstInterfaceWithMultipleGenericInstantiations with 
-    | None -> ()
-    | Some (typ1, typ2) -> 
-         errorR(Error(FSComp.SR.chkMultipleGenericInterfaceInstantiations((NicePrint.minimalStringOfType cenv.denv typ1), (NicePrint.minimalStringOfType cenv.denv typ2)), m))
+    if not(cenv.g.langVersion.SupportsFeature LanguageFeature.InterfacesWithMultipleGenericInstantiation) then
+        let table = interfaces |> MultiMap.initBy keyf
+        let firstInterfaceWithMultipleGenericInstantiations = 
+            interfaces |> List.tryPick (fun typ1 -> 
+                table |> MultiMap.find (keyf typ1) |> List.tryPick (fun typ2 -> 
+                       if // same nominal type
+                           tyconRefEq cenv.g (tcrefOfAppTy cenv.g typ1) (tcrefOfAppTy cenv.g typ2) &&
+                           // different instantiations
+                           not (typeEquivAux EraseNone cenv.g typ1 typ2) 
+                        then Some (typ1, typ2)
+                        else None))
+        match firstInterfaceWithMultipleGenericInstantiations with 
+        | None -> ()
+        | Some (typ1, typ2) -> 
+            let typ1Str = NicePrint.minimalStringOfType cenv.denv typ1
+            let typ2Str = NicePrint.minimalStringOfType cenv.denv typ2
+            errorR(Error(FSComp.SR.chkMultipleGenericInterfaceInstantiations(typ1Str, typ2Str), m))
+    else
+        let groups = interfaces |> List.groupBy keyf
+        let errors = seq {
+            for (_, items) in groups do
+                for i1 in 0 .. items.Length - 1 do
+                    for i2 in i1 + 1 .. items.Length - 1 do
+                        let typ1 = items.[i1]
+                        let typ2 = items.[i2]
+                        let tcRef1 = tcrefOfAppTy cenv.g typ1
+                        match compareTypesWithRegardToTypeVariablesAndMeasures cenv.g cenv.amap m typ1 typ2 with
+                        | ExactlyEqual -> () // exact duplicates are checked in another place
+                        | FeasiblyEqual ->
+                            let typ1Str = NicePrint.minimalStringOfType cenv.denv typ1
+                            let typ2Str = NicePrint.minimalStringOfType cenv.denv typ2
+                            if isObjectExpression then
+                                yield (Error(FSComp.SR.typrelInterfaceWithConcreteAndVariableObjectExpression(tcRef1.DisplayNameWithStaticParametersAndUnderscoreTypars, typ1Str, typ2Str),m))
+                            else
+                                let typStr = NicePrint.minimalStringOfType cenv.denv typ
+                                yield (Error(FSComp.SR.typrelInterfaceWithConcreteAndVariable(typStr, tcRef1.DisplayNameWithStaticParametersAndUnderscoreTypars, typ1Str, typ2Str),m))
+                        | NotEqual -> ()
```

Comment:
Somewhere in here we can check to see if there is a case of multiple generic interfaces being used, then we can call `tryLanguageFeatureErrorRecover cenv.g.langVersion LanguageFeature.InterfacesWithMultipleGenericInstantiation m`

Reply:
I will

> Somewhere in here we can check to see if there is a case of multiple generic interfaces being used, then we can call:
> 
> ```fsharp
> tryLanguageFeatureErrorRecover cenv.g.langVersion LanguageFeature.InterfacesWithMultipleGenericInstantiation m
> ```

I will take a look.